katdal/datasources.py

```diff
@@ -220,38 +246,44 @@ class TelstateDataSource(DataSource):
 
     Parameters
     ----------
-    telstate: :class:`katsdptelstate.TelescopeState` object
+    telstate : :class:`katsdptelstate.TelescopeState` object
         Telescope state with appropriate views
+    chunk_store : :class:`katdal.ChunkStore` object, optional
+        Alternative chunk store (defaults to S3 store specified in telstate)
     source_name : string, optional
         Name of telstate source (used for metadata name)
     """
-    def __init__(self, telstate, source_name='telstate'):
+    def __init__(self, telstate, chunk_store=None, source_name='telstate'):
         self.telstate = telstate
         # Collect sensors
         sensors = {}
         for key in telstate.keys():
             if not telstate.is_immutable(key):
-                sensors[key] = TelstateSensorData(telstate, key)
+                sensor_name = _shorten_key(telstate, key)
+                if sensor_name:
```

Comment:
I don't expect it to happen, but what if the sensor name is itself a prefix? In that case `sensor_name` will be the empty string here, and hence falsey.

Reply:
I'll happily ignore those pathological sensors with the current code.